route rule: Add support of `iif` and `action`.

cathay4t opened the pull request:
* Added `RouteRule.iif` for matching on incoming interface name.
* Added `RouteRule.action` for these actions:
  * `RouteRuleAction::Blackhole`
  * `RouteRuleAction::Unreachable`
  * `RouteRuleAction::Prohibit`

Bumped required of nispor to 1.2.9 for the fix of route rule actions.

Integration test cases included.

ffmancera commented on rust/src/lib/route_rule.rs:
```diff
@@ -329,12 +335,15 @@ impl RouteRuleEntry {
             log::error!("{}", e);
             return Err(e);
         }
+
```
leftover/not needed

ffmancera commented on rust/src/lib/nm/nm_dbus/connection/route_rule.rs:
```diff
@@ -144,3 +161,41 @@ pub(crate) fn nm_ip_rules_to_value(
     Ok(zvariant::Value::Array(rule_values))
 }
+
+const RTN_BLACKHOLE: u8 = 6;
+const RTN_UNREACHABLE: u8 = 7;
+const RTN_PROHIBIT: u8 = 8;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+#[non_exhaustive]
+pub enum NmIpRouteRuleAction {
+    Blackhole,
+    Unreachable,
+    Prohibit,
+    Other(u8),
+}
+
+impl From<u8> for NmIpRouteRuleAction {
+    fn from(v: u8) -> Self {
+        match v {
+            RTN_BLACKHOLE => Self::Blackhole,
+            RTN_UNREACHABLE => Self::Unreachable,
+            RTN_PROHIBIT => Self::Prohibit,
+            _ => {
+                log::warn!("Unsupported IP route rule action {}", v);
+                Self::Other(v)
+            }
+        }
+    }
+}
+
+impl From<NmIpRouteRuleAction> for u8 {
+    fn from(v: NmIpRouteRuleAction) -> Self {
+        match v {
+            NmIpRouteRuleAction::Blackhole => RTN_BLACKHOLE,
+            NmIpRouteRuleAction::Unreachable => RTN_UNREACHABLE,
+            NmIpRouteRuleAction::Prohibit => RTN_PROHIBIT,
+            NmIpRouteRuleAction::Other(d) => d,
+        }
+    }
```
This look good to me but considering `NmIpRouteRule` is something internal maybe we could have action as `Option<u8>` directly, I am fine with this, just a suggestion.

cathay4t replied:
The hope is someday we isolate `nm_dbus` to a public crate.

cathay4t committed:
route rule: Add support of `iif` and `action`.

* Added `RouteRule.iif` for matching on incoming interface name.
* Added `RouteRule.action` for these actions:
    * `RouteRuleAction::Blackhole`
    * `RouteRuleAction::Unreachable`
    * `RouteRuleAction::Prohibit`

Bumped required of nispor to 1.2.9 for the fix of route rule actions.

Integration test cases included.

Signed-off-by: Gris Ge <fge@redhat.com>
